### DISCUSSION TITLE
What does this cost?

### DISCUSSION THREAD

**Discussion by KCErb** [mebs/CM4-NAS#8] [Q&A]
I'm thinking about using this project to get into the world of electronics prototyping! I don't know anything about getting all these little components, getting a PCB printed, mounting everything and so on.

What do these things cost roughly?

I see a parts list and some tools are mentioned (like a reflow oven) so I can google those (is a microscope necessary to do it properly?). And it looks like I can send the fab to a company and get a quote. But it'd be nice for anyone who has actually done this to post information about quotes they got, tools they used, parts they bought and so on. It'd be convenient to, at a glance, see the cost in terms of money but also other factors like shipping times and so on.

If people from different countries could chime in as well that might help a beginner like myself get some idea of what they're getting themselves into here :)

**Comment by aschlean** [comment]
Hi!

Depends on where you live, but the easiest way to get components would be from distributors.
Most renowned are Digikey in the US, TME/Mouser in Europe, and Farnell in the UK.

As per PCBs, the fastest and cheapest way would be to order from JLCPBC or PCBWay. It's super easy to get a quote from them, and if you pay for shipping you can have them at your doorstep really fast. You can also order already assembled PCBs. You can learn more about that on their sites:
https://jlcpcb.com/
https://www.pcbway.com/

If you want to have a reflow oven, which definitely makes life easier, here is a link to a project from Unexpected Maker that turns a regular toaster oven into a reflow:
https://www.tindie.com/products/seonr/reflow-master/
https://www.youtube.com/watch?v=B-J3ShrOQuI&t=54s

But, if you want to get into electronics and PCBs and you're not confident in messing with a toaster oven I think it's best for you to order PCBs assembled or buy a hot air station/soldering station and learn how to solder components by hand.

Hope I helped you with your questions!
aschlean

**Reply by KCErb** [reply]
That does help a bit, thanks for the links! 

But I think this Q&A would be helpful to others if it included at least an order of magnitude idea of cost. Is a custom-printed PCB ~$10USD or $100USD? Can you print just 1? 

(When I go on the website I see things like "starting at $5" but I don't know how that compares to this NAS project. I need dimensions, layers, thickness and so on. To price out the assembly there are loads more questions too! I'm excited to learn what this all means but before I dive that deep it'd be nice to have some idea what I'm diving into.)

And as for equipment, I can see some cheap microscopes, oscilloscopes, soldering stations, airflow stations, etc. some places online but I have no idea what to expect in terms of quality, or whether these are a must-have or what criteria to judge them by.

Of course, so much information probably doesn't belong here, but if someone could post a short list of stuff they used to build this and about what it cost them in their country that would help nail down the estimate for someone like me a whole lot!

**Comment by mebs** [comment]
Hi there!

@aschlean's answer already has most of what you need to know to get started, but I can complement it with my own experience with this project and give you a rough idea of what you're getting into. It can vary a lot depending on your approach and how you value your time, though. 
Also note that all the costs I will list here are in Canadian dollars as that's where I live.

Excluding the tools and shipping for the parts, you could spend around 90CAD to get 5 units of rev1.4 of this PCB (assuming you don't mess any of them up):

- JLCPCB ships PCBs by batch of 5 and costs go up as your PCB gains in complexity (layers), cost me around 25$.
- I bought some of the parts from Digikey before I was confident enough to buy from LCSC (Chinese). A poor choice as most of the part are much more expensive there. The cheapest MagJack (ethernet jack with magnetics) I found on the website was 8$ and I designed the PCB with that footprint. So this represents 40$ of the total 90$ for the 5 units. If I ever manufacture the board I would change the design to use a much cheaper unit from LCSC.
- The rest of the parts are too many to list here but can all be bought on LCSS, including the high density connector.

So you could buy only one MagJack as I did and end up paying around 60$ for a single fully featured board. You would just solder it last once you know the rest works.

Now for the tooling. As aschlean said you can ask for PCB assembly from JLCPCB with your order, they work with LCSC (probably the same company group) to provide the parts. This would involve preparing some sort of descriptive files for the pick-and-place process, I didn't go down that route so I can't provide more info on that. There's obviously additional costs with assembly, but I've seen it often offered for free as a promotion.

If you want to do it yourself as I did, you need at the very least a:
- Hot air gun
- Soldering iron
- Good magnifier
- Multimeter for testing
- Bit of time and courage :)

This list doesn't include the consumables.

If you want to get started right away and you have none of the tooling, you'd better get a full blown soldering station. I pushed a bit and made my own as a project... for this project. 
If you want to do this for as cheap as possible, there are good enough soldering irons on Ali Express for less than 10$. You can also buy a hot air gun for 15$-20$. To make the air gun work you can use a regular incandescent light dimmer switch like the ones you might have on your walls. If you want to be able to read the air temperature, the gun contains a regular K-type thermocouple you can hookup to a multimeter that has this feature for example. It'll be a mess so you may want to make a nice box for all of that.

Edit: I agree with aschlean, just order the assembly!

Hope that helps!

**Reply by KCErb** [reply]
Oh wow, OK, thank you. I didn't guess it would be so cheap. I have a soldering station and magnifier + helping hands but no air gun. I'm also in Canada so that helps too!

I was originally thinking about doing my own PCB but quickly lost courage. Starting with yours though I'll dare to customize. So I'll make changing the ethernet jack my first quest and see where things go. Thanks!
